themes/westerndevs/scripts/author-list-json.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+hexo.extend.generator.register('author-list-json', function(locals) {
+
+  var authorList = [];
+  var authorKeys = Object.keys(locals.data.authors);
+  for (var i = 0; i < authorKeys.length; i++) {
```

Comment:
There are two weird for loop concepts in javascript. `for of` is the same as C#'s foreach while `for in` loops over properties. So if you `for in` an array you actually get the keys from the array. Behind the scenes all JS arrays are actually associative arrays so you get a bunch of keys which are just 1,2,3,4.